libyul/optimiser/LoopInvariantCodeMotion.cpp

```diff
@@ -82,6 +85,11 @@ optional<vector<Statement>> LoopInvariantCodeMotion::rewriteLoop(ForLoop& _for)
 {
 	assertThrow(_for.pre.statements.empty(), OptimizerException, "");
 	vector<Statement> replacement;
+
+	Block block{_for.location, {}};
+	block.statements.emplace_back(ASTCopier{}(_for));
```

Comment:
Why do you copy here?

Reply:
Because `ForLoop` does not have a copy constructor (the condition is a unique pointer.)

Reply:
Sure, but why can't you move?

Reply:
Because `_for.post` and `_for.body` is used afterwards. We could possibly move `_for`, and later get these from `block.statements` vector. But that would be less readable, I think.